sklearn/ensemble/forest.py

```diff
@@ -2009,7 +2009,9 @@ def fit_transform(self, X, y=None, sample_weight=None):
         super().fit(X, y, sample_weight=sample_weight)
 
         self.one_hot_encoder_ = OneHotEncoder(sparse=self.sparse_output)
-        return self.one_hot_encoder_.fit_transform(self.apply(X))
+        res = self.one_hot_encoder_.fit_transform(self.apply(X))
+        self.n_features_out_ = res.shape[1]
```

Comment:
This means most other classes pass `hasattr(instance, 'feature_names_out_')` even before `fit`, but this class would fail on that. I think it's better to be consistent on that.

Reply:
Which ones do? Wouldn't that break the ``not setting anything in __init__`` test? Though the clusterers and the components do have those before fit. Why doesn't that break. huh....